src/Features/CSharp/Portable/QuickInfo/CSharpDiagnosticAnalyzerQuickInfoProvider.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.QuickInfo;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.QuickInfo
+{
+    [ExportQuickInfoProvider(QuickInfoProviderNames.DiagnosticAnalyzer, LanguageNames.CSharp), Shared]
+    [ExtensionOrder(After = QuickInfoProviderNames.Syntactic)]
+    internal class CSharpDiagnosticAnalyzerQuickInfoProvider : CommonQuickInfoProvider
+    {
+        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpDiagnosticAnalyzerQuickInfoProvider(IDiagnosticAnalyzerService diagnosticAnalyzerService)
+        {
+            _diagnosticAnalyzerService = diagnosticAnalyzerService;
+        }
+
+        protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
+            Document document,
+            SyntaxToken token,
+            CancellationToken cancellationToken)
+        {
+            var (pragmaWarning, errorCode) = token.Parent switch
+            {
+                PragmaWarningDirectiveTriviaSyntax directive when directive.IsDisablePragma()
```

Comment:
Why only disable pragma and not restore?

Reply:
Because I call `GetDiagnosticsForSpanAsync` passing the text range starting from the end of the pragma (to do: until the enable pragma).

Reply:
I don't think you should force compute diagnostics from a quick info provider, especially given you don't really need the diagnostic here (for unnecessary pragma, there may not even be one). I would suggest removing this when clause, just fetch the diagnostic IDs for the pragma directive and use descriptors to map ID to display string.

Reply:
Done and added test `ErrorTileIsShownOnRestorePragma`